### PR TITLE
Canary

### PR DESCRIPTION
**Pull request by aintgoin2goa**
Created a new "canary" version of the front-page, which pivots on an env var rather than a flag.

This new version is deployed to a separate heroku app.   Later on I will figure out how to proxy to it from the real front-page

### CODE REVIEW

File: server/init.js [hunk 1]
```diff
@@ -1,2 +1,9 @@
 require('./setup');
-module.exports = require('./app');
+
+process.on('uncaughtException', err => {
```
Comment (aintgoin2goa) [inline server/init.js]: Appreciate this is a but I've some issues locally with uncaught exceptions and need to be sure it's ok on heroku

File: server/app-v2.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	withFlags: true,
+	withHandlebars: true,
+	withNavigation: true,
+	withAnonMiddleware: true,
+	withRequestTracing: true,
+	withBackendAuthentication: true,
+	hasHeadCss: true,
+	hasNUiBundle: true,
+	layoutsDir: path.join(process.cwd(), '/bower_components/n-ui/layout'),
+	// helpers: {}
```
Comment (leggsimon) [inline server/app-v2.js]: remove this?

File: server/controllers/front-page.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	const renderParams = {
+		layout: 'wrapper',
+		region: region.edition
+	};
+	setupMarketsHeader(res.locals.flags, renderParams);
+	res.render('front-page-v2', renderParams);
+}
+
+function ukHomePage(req, res){
+	let edition = 'UK';
```
Comment (leggsimon) [inline server/controllers/front-page.js]: `const`

File: server/init.js [hunk 4]
```diff
@@ -1,2 +1,9 @@
...
+	console.error(err.stack);
+});
+
+let app = process.env.CANARY ? require('./app-v2') : require('./app');
```
Comment (leggsimon) [inline server/init.js]: `const`

File: server/controllers/front-page.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+"use strict";
+let dataModel;
+
+const marketsDataTemplate =
+	`
+	<div class="markets-data-wrapper">
+		<div class="o-grid-container">
+			<div class="o-grid-row">
+				<div class="markets-data js-markets-data" data-o-grid-colspan="12" data-trackable="header | markets data">
```
Comment (i-like-robots) [inline server/controllers/front-page.js]: we can lose the `div.o-grid-row` element and `[data-o-grid-colspan="12"]` attribute

File: server/controllers/front-page.js (hunk 3 above)
Comment (i-like-robots) [inline server/controllers/front-page.js]: const

File: server/controllers/front-page.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+function ukHomePage(req, res){
+	let edition = 'UK';
+	const data = dataModel.ukHomePage;
+	return render(req, res, edition, data);
+}
+
+function internationalHomePage(req, res){
+	let edition = 'US';
```
Comment (i-like-robots) [inline server/controllers/front-page.js]: and here

File: server/controllers/front-page.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	const data = dataModel.internationalHomePage;
+	return render(req, res, edition, data);
+}
+
+module.exports = model => {
+	dataModel = model;
```
Comment (aintgoin2goa) [inline server/controllers/front-page.js]: It's used just above - but actually has been renamed to dataStore

File: server/app-v2.js (hunk 2 above)
Comment (aintgoin2goa) [inline server/app-v2.js]: That's to just to remind me to put them back later

**Pull request merged**